src/app/pages/storage/modules/vdevs/components/zfs-info-card/raidz-extend-dialog/raidz-extend-dialog.component.ts

```diff
@@ -73,26 +77,51 @@ export class RaidzExtendDialog {
   protected onSubmit(event: SubmitEvent): void {
     event.preventDefault();
 
-    const payload = {
-      // UI check for duplicate serials is handled in UnusedDiskSelectComponent
-      allow_duplicate_serials: true,
-      new_disk: this.form.value.newDisk,
-      target_vdev: this.data.vdev.guid,
-    } as PoolAttachParams;
+    // Check for existing pool.attach jobs for this pool
+    this.checkForExistingExtendJob().pipe(
+      switchMap((hasExistingJob) => {
+        if (hasExistingJob) {
+          this.snackbar.error(
+            this.translate.instant('A VDEV extension operation is already in progress for this pool. Please wait for it to complete.'),
+          );
+          return of(null);
+        }
 
-    this.dialogService.jobDialog(
-      this.api.job('pool.attach', [this.data.poolId, payload]),
-      { title: this.translate.instant('Extending VDEV'), canMinimize: true },
-    )
-      .afterClosed()
-      .pipe(
-        this.errorHandler.withErrorHandler(),
-        untilDestroyed(this),
-      )
-      .subscribe(() => {
+        const payload = {
+          // UI check for duplicate serials is handled in UnusedDiskSelectComponent
+          allow_duplicate_serials: true,
+          new_disk: this.form.value.newDisk,
+          target_vdev: this.data.vdev.guid,
+        } as PoolAttachParams;
+
+        return this.dialogService.jobDialog(
+          this.api.job('pool.attach', [this.data.poolId, payload]),
+          { title: this.translate.instant('Extending VDEV'), canMinimize: true },
+        ).afterClosed();
+      }),
+      this.errorHandler.withErrorHandler(),
+      untilDestroyed(this),
+    ).subscribe((result) => {
+      if (result) {
         this.snackbar.success(this.translate.instant('VDEV successfully extended.'));
         this.dialogRef.close(true);
-      });
+      }
+    });
+  }
+
+  private checkForExistingExtendJob(): Observable<boolean> {
```

Comment:
seems we have duplicated logic here. Should we use a service to avoid duplication?